frontends/PyCDE/src/pycde/constructs.py

```diff
@@ -113,22 +113,26 @@ def __setitem__(self, idxOrSlice: Union[int, slice], value):
 
 
 def Reg(type: PyCDEType,
-        clk: Value = None,
-        rst: Value = None,
-        rst_value: Value = None):
+        clk: PyCDEValue = None,
+        rst: PyCDEValue = None,
+        rst_value=None,
+        ce: PyCDEValue = None):
   """Declare a register. Must assign exactly once."""
 
   class RegisterValue(type._get_value_class()):
 
-    def assign(self, new_value: Value):
+    def assign(self, new_value: PyCDEValue):
       if self._wire is None:
         raise ValueError("Cannot assign value to Reg twice.")
       self._wire.assign(new_value)
       self._wire = None
 
   # Create a wire and register it.
   wire = Wire(type)
-  value = RegisterValue(wire.reg(clk=clk, rst=rst, rst_value=rst_value), type)
+  if rst_value is not None and not isinstance(rst_value, PyCDEValue):
```

Comment:
nit: i personally prefer `if not (rst_value and isinstance(rst_value, PyCDEValue))` for readability.

Reply:
IMO that obscures the intent. The way to read this is "if there is a reset value specified and it's not a PyCDE value", (next line) "try to make it a PyCDE value". That's not immediately obvious with the expression you specified (as evidenced by the `and` should be an `or`).